### PR TITLE
Consolidate ve configuration

### PR DESCRIPTION
**Pull request by haoshen61**

### CODE REVIEW

**Comment by zhang-shengping** [comment]
you probably need to merge these commits to one.

File: app/waf/src/controllers/adc.controller.ts [hunk 1]
```diff
@@ -579,10 +580,20 @@ export class AdcController extends BaseController {
       .get(WafBindingKeys.KeyNetworkDriver)
       .then(async networkHelper => {
         for (let k of Object.keys(adc.networks)) {
+          if (!adc.management) {
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: How about to move L583-590 outside the `for`. 
And replace it with `merge(adc.management, {networks: {}});`.

File: app/waf/src/controllers/adc.controller.ts [hunk 2]
```diff
@@ -579,10 +580,20 @@ export class AdcController extends BaseController {
...
+          if (!adc.management.networks) {
+            adc.management.networks = {};
+          }
           let net = adc.networks[k];
-          if (net.portId && net.ready) continue;
-
-          net.ready = false;
+          if (adc.management.networks[k] && adc.management.networks[k].portId) {
+            adc.management.networks[k].fixedIp = net.fixedIp;
+            adc.management.networks[k].macAddr = net.macAddr;
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: macAddr is generated after `createPort`.

File: app/waf/src/controllers/adc.controller.ts [hunk 3]
```diff
@@ -593,10 +604,10 @@ export class AdcController extends BaseController {
           await networkHelper
             .createPort(addon.userToken, portParams)
             .then(async port => {
-              net.fixedIp = port.fixedIp;
-              net.macAddr = port.macAddr;
-              net.portId = port.id;
-              net.ready = true;
+              adc.management.networks[k] = {};
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: We may combine these value assignments.

File: app/waf/src/controllers/adc.controller.ts [hunk 4]
```diff
@@ -672,13 +680,13 @@ export class AdcController extends BaseController {
         let computeMgr = await this.wafapp.get(
           WafBindingKeys.KeyComputeManager,
         );
-        if (adc.compute.vmId) {
+        if (adc.management.connection && adc.management.vmId) {
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: really need `adc.management.connection &&`?

File: app/waf/src/models/adc.model.ts [hunk 5]
```diff
@@ -41,26 +41,35 @@ export type ConfigTypes = {
...
+    vmId?: string; // cannot be appointed.
+    networks: {
+      [key: string]: {
+        fixedIp?: string;
+        macAddr?: string;
+        //floatingIp?: string;
+        portId?: string; // cannot be appointed.
+        //vips?: [string]; // cannot be appointed.
+      };
+    };
```
Comment (zongzw) [inline app/waf/src/models/adc.model.ts]: trustedDeviceId?

Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: Leave it, and will handle it later.

**Pull request merged**